### ISSUE TITLE
Haiku: Can't find Python.h header

### ISSUE THREAD

**Issue by roired** [Nuitka/Nuitka#1314]
Good day,

I'm a Python newbie and first time Nuitka user, have no idea of GCC nor Clang nor SCons. Trying to run nuitka inside a VENV:
(penv) /Projects/Testing/pythonTesting/testnui> python -m nuitka --version
0.6.18.2
Commercial: None
Python: 3.8.12 (default, Nov 20 2021, 23:55:23) 
Flavor: Unknown
Executable: /Projects/Testing/pythonTesting/testnui/penv/bin/python
OS: Haiku
Arch: x86_64

Nuitka was installed inside the VENV with 'python -m pip install nuitka'.

Inside the VENV only the following modules are present:
(penv) /Projects/Testing/pythonTesting/testnui> python -m pip list
Package    Version
---------- --------
Nuitka     0.6.18.2
pip        21.3.1
PyOpenGL   3.1.5
PyQt5      5.15.6
PyQt5-sip  12.9.0
setuptools 56.0.0

The VENV was created with "python3.8 -m venv penv --system-site-packages'

I'm trying to run nuitka on a very simple PyQt script:
```
import sys
from PyQt5.QtWidgets import QApplication, QWidget
if __name__ == "__main__":
    app = QApplication(sys.argv)
    w = QWidget()
    w.resize(300,300)
    w.setWindowTitle('test')
    w.show()
    sys.exit(app.exec_())
```
The example runs ok with the python interpreter, but when I try to run the nuitka module I get an error:
```
(penv) /Projects/Testing/pythonTesting/testnui> python -m nuitka test.py 
Nuitka-Options:INFO: Used command line options: test.py
Nuitka:INFO: Starting Python compilation with Nuitka '0.6.18.2' on Python '3.8' commercial None.
Nuitka:INFO: Completed Python level compilation and optimization.
Nuitka:INFO: Generating source code for C backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.
Nuitka:INFO: Running C compilation via Scons.
Nuitka-Scons:INFO: Backend C compiler: gcc (gcc).
FATAL: Error, no 'Python.h' development headers can be found at '['/boot/system/include/python3.8']', dependency not satisfied!
  
```

On the system I have installed GCC 11.2.0 and llvm12_clang 12.0.0-1, and SCons 3.1.2-3.

Haiku does not have a /boot/system/include, it has a /boot/system/develop/headers where the Python headers are located. 
[edited: so I presume just change/add the /boot/system/develop/headers/python3.8 would be enough to fix this?. In which file should be replaced?]

I would appreciate any help regarding this issue.
Thanks for your assistance.

Regards,
RR

**Comment by roired** [comment]
Good day,

I did some digging through the files and found the location of the path. It's the file Backend.scons. The method _detectPythonHeaderPath. Added the Haiku path to Python3.8 header files and it succeeded. Should try with a real program, not with a simple test and might be good to package it for Haiku too.

Regards,
RR

**Comment by kayhayen** [comment]
@roired I have no idea what Haiku is an OS, but BeOS was always an impressive idea. You are welcome to add something that creates the path from the `python_prefix` there, for everything non-Windows, the list currently is this:

```
        candidates = [
            os.path.join(python_prefix, "include", "python" + python_abi_version)
        ]
```

Please try make a PR out of it, or send me a diff this way if that's too hard for you.

**Comment by roired** [comment]
Good day @kayhayen ,

Haiku is the successor of BeOS and is being actively developed, and yes, Haiku still has the same feeling BeOS had back in the day, and I'm using it on a daily basis, that is why I needed something like Nuitka :) to do some Python stuff. When I test with a real application we will see if we need to modify anything else, but as of now, the 'Backend.scons' seems to be the only one needed.

I'll try to make a pull request to add the line, to Backend.scons. If I can't get it done I'll post the line here, nonetheless, I should try and succeed at doing the PR. Never too late to learn. Right now I just added the hardcoded path to 'Backend.scons' so have to change it to make it independent of the Python version. Once I had that tested I'll get on with the PR.

It's good that I could install Nuitka through PIP without any issues, and got it working. The fact that Nuitka dependencies are so few helped a lot to get this working on Haiku.

Thanks!
Regards,
RR

**Comment by roired** [comment]
Good day @kayhayen,

I tried to do the PR today, not sure if succeeded. I just added a comment line and the Haiku OS path to the Python header.

I did some test with the --follow-imports option on VM without success, have to test this again because Haiku VM inside Haiku OS don't perform well. I might have issues with QEMU. I'll try on metal and see if can build a more complex app. For single file Nuitka does perform quite well on Haiku.

Regards,
RR
